### DISCUSSION TITLE
Which is the alternative hook for useDapp `useTokenAllowance`

### DISCUSSION THREAD

**Discussion by zouhangwithsweet** [wevm/wagmi#685] [Q&A]
I am migrating my dapp to wagmi. And I don't know the alternative hook for https://usedapp-docs.netlify.app/docs/API%20Reference/Hooks#usetokenallowance

**Comment by tmm** [comment]
There isn't a dedicated hook for this, but the following should work.

```ts
import { erc20ABI, useContracRead } from 'wagmi'

useContractRead({
  addressOrName: tokenAddress,
  contractInterface: erc20ABI,
  functionName: 'allowance',
  args: [ownerAddress, spenderAddress],
})
```

**Reply by zouhangwithsweet** [reply]
Thanks very much

**Comment by smakosh** [comment]
I opened a new discussion about this https://github.com/wagmi-dev/wagmi/discussions/1036
